### PR TITLE
[StateService] More RPC methods and commands

### PR DESCRIPTION
**Pull request by ZhangTao1596**
### Step 3/3 StateService
Since there are too many changes in StateService, I will split it into 3 parts to commit pull request, and this is the 2nd.
  - [x] Storage and P2P message (Merged)
  - [x] Verification
  - [x] More command & RPC
  
Changes:
* Add `GetStateRoot` `GetProof` `VerifyProof` RPC methods.
* Add console commands `state root [index]` `state height`

### CODE REVIEW

File: src/StateService/StatePlugin.cs [hunk 1]
```diff
@@ -37,5 +46,92 @@ void IPersistencePlugin.OnPersist(Block block, StoreView snapshot, IReadOnlyList
...
+            using var snapshot = StateStore.Singleton.GetSnapshot();
+            StateRoot state_root = snapshot.GetStateRoot(index);
+            if (state_root is null)
+                throw new RpcException(-100, "Unknown state root hash");
+            else
+                return state_root.ToJson();
+        }
+
+        [RpcMethod]
+        public JObject GetProof(JArray _params)
```
Comment (shargon) [inline src/StateService/StatePlugin.cs:80]: `GetProof` and `VerifyProof` for console too?

**Comment by shargon** [comment]
Tested?

**Review by shargon — APPROVED** [review]

**Comment by ZhangTao1596** [comment]
> Tested?

@superboyiii is testing.

**Review by shargon — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Comment by superboyiii** [comment]
I registered and started state service successfully, but later state roots are still don't have witness. 
image

**Comment by ZhangTao1596** [comment]
> I registered and started state service successfully, but later state roots are still don't have witness.
> image
> And ValidatedRootIndex is null as well.
> image

You should comment under https://github.com/neo-project/neo-modules/pull/470

File: src/StateService/StatePlugin.cs [hunk 2]
```diff
@@ -36,5 +44,128 @@ void IPersistencePlugin.OnPersist(Block block, DataCache snapshot, IReadOnlyList
...
+        }
+
+        [RpcMethod]
+        public JObject GetStateRoot(JArray _params)
+        {
+            uint index = uint.Parse(_params[0].AsString());
+            using var snapshot = StateStore.Singleton.GetSnapshot();
+            StateRoot state_root = snapshot.GetStateRoot(index);
+            if (state_root is null)
+                throw new RpcException(-100, "Unknown state root hash");
```
Comment (superboyiii) [inline src/StateService/StatePlugin.cs:98]:
```suggestion
                throw new RpcException(-100, "Unknown state root");
```

Comment (ZhangTao1596) [reply src/StateService/StatePlugin.cs]: Done.

**Comment by superboyiii** [comment]
I've totaly tested all of this PR, it works well on mine. Able to merge.

**Review by superboyiii — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**